src/Files.App/Filesystem/FilesystemOperations/ShellFilesystemOperations.cs

```diff
@@ -764,20 +682,10 @@ public async Task<IStorageHistory> RestoreItemsFromTrashAsync(IList<IStorageItem
             connection.RequestReceived += handler;
 
             var moveResult = new ShellOperationResult();
-            var (status, response) = await connection.SendMessageForResponseAsync(new ValueSet()
-            {
-                { "Arguments", "FileOperation" },
-                { "fileop", "MoveItem" },
-                { "operationID", operationID },
-                { "filepath", string.Join('|', source.Select(s => s.Path)) },
-                { "destpath", string.Join('|', destination) },
-                { "overwrite", false },
-                { "HWND", NativeWinApiHelper.CoreWindowHandle.ToInt64() }
-            });
-            var result = (FilesystemResult)(status == AppServiceResponseStatus.Success
-                && response.Get("Success", defaultJson).GetBoolean());
-            var shellOpResult = JsonSerializer.Deserialize<ShellOperationResult>(response.Get("Result", defaultJson).GetString());
-            moveResult.Items.AddRange(shellOpResult?.Final ?? Enumerable.Empty<ShellOperationItemResult>());
+            var (status, response) = await FileOperationsHelpers.MoveItemAsync(string.Join('|', source.Select(s => s.Path)), string.Join('|', destination), false, NativeWinApiHelper.CoreWindowHandle.ToInt64(), operationID);
```

Comment:
Given that the interop between Files.App and Files.FullTrust is no longer needed, we shouldn't serialize paths to a string anymore. Instead we should pass the `string[]` for paths directly.